pmdarima/arima/_auto_solvers.py

```diff
@@ -105,55 +161,77 @@ def is_new_better(self, new_model):
+        if not self.seasonal:
+            seasonal_order = (0, 0, 0, 0)
 
-        order = (p, self.d, q)
-        ssnl = (P, self.D, Q, self.m) if self.seasonal else (0, 0, 0, 0)
+        # we might be fitting without a constant
+        if constant is None:
+            constant = self.with_intercept
 
-        # if the sum of the orders > max_order we do not build this model...
-        order_sum = p + q + (P + Q if self.seasonal else 0)
+        if (order, seasonal_order, constant) not in self.results_dict:
 
-        # if all conditions hold true, good to build this model.
-        if expr and order_sum <= self.max_order and (order, ssnl) not \
-                in self.results_dict:
+            # increment the number of fits
             self.k += 1
-            # cache the fit
-            fit = _fit_arima(self.y, xreg=self.xreg, order=order,
-                             seasonal_order=ssnl,
-                             start_params=self.start_params, trend=self.trend,
-                             method=self.method, transparams=self.transparams,
-                             solver=self.solver, maxiter=self.maxiter,
-                             disp=self.disp, callback=self.callback,
-                             fit_params=self.fit_params,
-                             suppress_warnings=self.suppress_warnings,
-                             trace=self.trace, error_action=self.error_action,
-                             out_of_sample_size=self.out_of_sample_size,
-                             scoring=self.scoring,
-                             scoring_args=self.scoring_args,
-                             with_intercept=self.with_intercept,
-                             **self.kwargs)
+
+            fit = self._fit_arima(
+                order=order,
+                seasonal_order=seasonal_order,
+                with_intercept=constant)
 
             # use the orders as a key to be hashed for
             # the dictionary (pointing to fit)
-            self.results_dict[(order, ssnl)] = fit
+            self.results_dict[(order, seasonal_order, constant)] = fit
 
             if self.is_new_better(fit):
                 self.bestfit = fit
+                return True
+        return False
 
-                # reset p, d, P, Q, etc.
-                self.p, self.q, self.P, self.Q = p, q, P, Q
-
-    def step_through(self):
+    def solve_stepwise(self):
         start_time = datetime.now()
+        p, d, q = self.p, self.d, self.q
+        P, D, Q, m = self.P, self.D, self.Q, self.m
+        max_p, max_q = self.max_p, self.max_q
+        max_P, max_Q = self.max_P, self.max_Q
+
+        # fit a baseline p, d, q model
+        self._do_fit((p, d, q), (P, D, Q, m))
+
+        # null model with possible constant
+        if self._do_fit((0, d, 0), (0, D, 0, m)):
+            p = q = P = Q = 0
+
+        # A basic AR model
+        if max_p > 0 or max_P > 0:
+            _p = int(max_p > 0)
```

Comment:
I feel like these are a little more readable like this:

```python
_p = 1 if max_p > 0 else 0
```

It's also a little faster:

```python
>>> import timeit
>>> timeit.timeit("max_p = 1; _p = int(max_p > 0)", number=100000)
0.018097285999999713
>>> timeit.timeit("max_p = 1; _p = 1 if max_p > 0 else 0", number=100000)
0.005227653000019927
```

Thoughts?

Reply:
Pretty amazing that there is such a time differential. Even as minute as it really is... will change

Reply:
I thought it might be because it didn't have to evaluate the second half of the expression, but it is just as fast using `max_p = 0`:

```python
>>> import timeit
>>> timeit.timeit("max_p = 0; _p = int(max_p > 0)", number=100000)
0.017676304000019627
>>> timeit.timeit("max_p = 0; _p = 1 if max_p > 0 else 0", number=100000)
0.004342629000007037
```